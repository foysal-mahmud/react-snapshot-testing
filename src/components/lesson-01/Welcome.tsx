import React from 'react';

interface WelcomeProps {
  name?: string;
  showGreeting?: boolean;
}

const Welcome: React.FC<WelcomeProps> = ({ 
  name = 'Guest', 
  showGreeting = true 
}) => {
  return (
    <div className="welcome-container" data-testid="welcome-container">
      {showGreeting && (
        <h1>Welcome, {name}!</h1>
      )}
      <p>Thank you for joining our React Snapshot Testing course.</p>
      <div className="info" data-testid="date-info">
        <span>Current date: {new Date().toLocaleDateString()}</span>
      </div>
    </div>
  );
};

export default Welcome; 