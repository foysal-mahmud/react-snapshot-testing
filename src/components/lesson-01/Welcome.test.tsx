import React from 'react';
import { render, screen } from '@testing-library/react';
import Welcome from './Welcome';

describe('Lesson 1: Welcome Component', () => {
  // Basic rendering test
  test('renders welcome message with default props', () => {
    render(<Welcome />);
    
    const welcomeHeading = screen.getByRole('heading', { level: 1 });
    expect(welcomeHeading).toHaveTextContent('Welcome, Guest!');
    
    const courseMessage = screen.getByText(/thank you for joining our react snapshot testing course/i);
    expect(courseMessage).toBeInTheDocument();
  });

  // Testing with props
  test('renders welcome message with custom name', () => {
    render(<Welcome name="John" />);
    
    const welcomeHeading = screen.getByRole('heading', { level: 1 });
    expect(welcomeHeading).toHaveTextContent('Welcome, John!');
  });

  // Testing conditional rendering
  test('does not render greeting when showGreeting is false', () => {
    render(<Welcome showGreeting={false} />);
    
    const welcomeHeading = screen.queryByRole('heading', { level: 1 });
    expect(welcomeHeading).not.toBeInTheDocument();
    
    // But the course message should still be there
    const courseMessage = screen.getByText(/thank you for joining our react snapshot testing course/i);
    expect(courseMessage).toBeInTheDocument();
  });

  // Testing element classes and structure
  test('renders with correct structure and data attributes', () => {
    render(<Welcome />);
    
    const container = screen.getByTestId('welcome-container');
    expect(container).toBeInTheDocument();
    expect(container).toHaveClass('welcome-container');
    
    const dateInfo = screen.getByTestId('date-info');
    expect(dateInfo).toBeInTheDocument();
    expect(dateInfo).toHaveClass('info');
  });

  // LESSON 1 SNAPSHOT TESTS - Your introduction to snapshot testing! ðŸŽ‰
  describe('Lesson 1: Introduction to Snapshot Testing', () => {
    test('matches snapshot with default props', () => {
      render(<Welcome />);
      const welcomeComponent = screen.getByTestId('welcome-container');
      expect(welcomeComponent).toMatchSnapshot();
    });

    test('matches snapshot with custom props', () => {
      render(<Welcome name="Alice" showGreeting={false} />);
      const welcomeComponent = screen.getByTestId('welcome-container');
      expect(welcomeComponent).toMatchSnapshot();
    });

    test('matches snapshot for lesson 1 student', () => {
      render(<Welcome name="Student" />);
      const welcomeComponent = screen.getByTestId('welcome-container');
      expect(welcomeComponent).toMatchSnapshot();
    });
  });
}); 