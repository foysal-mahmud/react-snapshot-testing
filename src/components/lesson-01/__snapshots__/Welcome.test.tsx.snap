// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lesson 1: Welcome Component Lesson 1: Introduction to Snapshot Testing matches snapshot for lesson 1 student 1`] = `
<div
  class="welcome-container"
  data-testid="welcome-container"
>
  <h1>
    Welcome, 
    Student
    !
  </h1>
  <p>
    Thank you for joining our React Snapshot Testing course.
  </p>
  <div
    class="info"
    data-testid="date-info"
  >
    <span>
      Current date: 
      5/23/2025
    </span>
  </div>
</div>
`;

exports[`Lesson 1: Welcome Component Lesson 1: Introduction to Snapshot Testing matches snapshot with custom props 1`] = `
<div
  class="welcome-container"
  data-testid="welcome-container"
>
  <p>
    Thank you for joining our React Snapshot Testing course.
  </p>
  <div
    class="info"
    data-testid="date-info"
  >
    <span>
      Current date: 
      5/23/2025
    </span>
  </div>
</div>
`;

exports[`Lesson 1: Welcome Component Lesson 1: Introduction to Snapshot Testing matches snapshot with default props 1`] = `
<div
  class="welcome-container"
  data-testid="welcome-container"
>
  <h1>
    Welcome, 
    Guest
    !
  </h1>
  <p>
    Thank you for joining our React Snapshot Testing course.
  </p>
  <div
    class="info"
    data-testid="date-info"
  >
    <span>
      Current date: 
      5/23/2025
    </span>
  </div>
</div>
`;
